packages/flutter/lib/src/widgets/_web_image_web.dart

```diff
@@ -0,0 +1,344 @@
+                const SizedBox.shrink(),
+        };
+        return builtWidget ?? child!;
+      },
+      child: FittedBox(
+        fit: BoxFit.scaleDown,
+        child: SizedBox(
+          width: naturalWidth,
+          height: naturalHeight,
+          child: HtmlElementView(
+            viewType: _ImgElementImageState._viewType,
+            creationParams: <String, String>{'src': widget.provider.src},
+            hitTestBehavior: PlatformViewHitTestBehavior.transparent,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _ImgElementProviderCache {
+  static const int _maxImages = 1000;
+  static const int _maxSizeInBytes = 100000000;
+  static final Map<String, _ImgElementProviderImpl> _imageCache =
+      <String, _ImgElementProviderImpl>{};
+  static final Queue<String> _imageQueue = Queue<String>();
+
+  static _ImgElementProviderImpl _getOrCreateProvider(String url) {
+    _ImgElementProviderImpl? info = _imageCache[url];
+    if (info == null) {
+      info = _ImgElementProviderImpl(url);
+      void Function()? imgLoadListener;
+      imgLoadListener = () {
+        // The image loaded, so we need to clean up the cache now that we know
+        // its size.
+        _cleanupCache();
+        info!.state.removeListener(imgLoadListener!);
+      };
+      info.state.addListener(imgLoadListener);
+      _imageCache[url] = info;
+    }
+    _touch(url);
+    return info;
+  }
+
+  static void _touch(String url) {
+    _imageQueue.remove(url);
+    _imageQueue.addFirst(url);
+  }
+
+  /// Remove the least recently used items from the cache until the number of
+  /// items in the cache is less than [_maxImages] and the total bytes of the
+  /// images in the cache is less than [_maxSizeInBytes].
+  static void _cleanupCache() {
+    assert(_imageQueue.length == _imageCache.length);
+    if (_imageQueue.length > _maxImages) {
+      final int imagesToRemove = _imageQueue.length - _maxImages;
+      for (int i = 0; i < imagesToRemove; i++) {
+        final String url = _imageQueue.removeLast();
+        final _ImgElementProviderImpl? info = _imageCache.remove(url);
+        if (info != null) {
+          if (info.state.value == _ImageLoadingState.loading) {
+            info.state.dispose();
+          }
+        }
+      }
+    }
+
+    double totalSize = 0;
+    for (final String url in _imageQueue) {
+      final _ImgElementProviderImpl? info = _imageCache[url];
+      totalSize += info!.estimatedSize;
+    }
+    while (totalSize > _maxSizeInBytes) {
+      final String url = _imageQueue.removeLast();
+      final _ImgElementProviderImpl? info = _imageCache.remove(url);
+      if (info != null) {
+        if (info.state.value == _ImageLoadingState.loading) {
+          info.state.dispose();
+        }
+        totalSize -= info.estimatedSize;
+      }
+    }
+  }
+}
+
+web.HTMLImageElement _createImage(String url) {
+  final web.HTMLImageElement image =
+      web.document.createElement('img') as web.HTMLImageElement;
```

Comment:
@harryterkelsen final web.HTMLImageElement image = web.HTMLImageElement();

Reply:
The constructor cannot be directly invoked.